models/detr.py

```diff
@@ -58,11 +58,20 @@ def forward(self, samples: NestedTensor):
         """
         if isinstance(samples, (list, torch.Tensor)):
             samples = nested_tensor_from_tensor_list(samples)
-        features, pos = self.backbone(samples)
+
+        # embed image
+        features, pos        = self.backbone(samples)
+
+        # embed query
+        query_fts, query_pos = self.backbone(query)
+
```

Comment:
Don't you have to pool the query somehow?  I didn't look closely at their `backbone`, but I think it returns the full feature map, not the global average pooled version which I'd think we want to use